packages/amplify-provider-awscloudformation/src/graphql-transformer/amplify-graphql-resource-manager.ts

```diff
@@ -142,31 +139,35 @@ export class GraphQLResourceManager {
       const tableNames = [];
       tables.forEach(tableName => {
         tableNames.push(tableNameMap.get(tableName));
-        const filepath = path.join(stateFileDir, stepNumber, 'stacks', `${tableName}.json`);
+        const filepath = path.join(stepPath, 'stacks', `${tableName}.json`);
         fs.ensureDirSync(path.dirname(filepath));
         JSONUtilities.writeJson(filepath, this.templateState.pop(tableName));
       });
 
       const deploymentRootKey = `${ROOT_APPSYNC_S3_KEY}/${buildHash}/states/${stepNumber}`;
       const deploymentStep: DeploymentOp = {
         stackTemplatePathOrUrl: `${deploymentRootKey}/cloudformation-template.json`,
+        prevMetaKey: prevMetaKey,
         parameters: { ...parameters, S3DeploymentRootKey: deploymentRootKey },
         stackName: this.resourceMeta.stackId,
         tableNames: tableNames,
         capabilities,
         // clientRequestToken: `${buildHash}-step-${stepNumber}`,
       };
 
+      // save the current deployment step in the state
+      const deploymentStateStep = path.join(stepPath, DEPLOYMENT_META);
```

Comment:
`deploymentStepStatePath`, based on usage it is the path of the state of the deployment step.